### PR TITLE
Support to ignore subdomain when engine pool conditions are met

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Support to ignore subdomain when engine pool conditions are met.
It is difficult to understand the engine routing rule for customers, especially for the engine pool and subdomain.
Maybe a user use subdomain before, and then they want to use engine pool.
Then specify the engine pool name, engine pool size, but the result is not expected.
They need to remove the subdomain specified before, otherwise, the engine pool will not work.
I need tell customer that, the engine subdomain config has higher priority and tell them the routing rule.
The understanding cost is high.
With this change, it is clear for engine pool conditions.
### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
how do you think about? @yaooqinn  @pan3793

**Comment by pan3793** [comment]
what's the background of this change?

**Comment by turboFei** [comment]
> what's the background of this change?


It is difficult to understand the engine routing rule for customers, especially for the engine pool and subdomain.


Maybe a user use subdomain before, and then they want to use engine pool.

Then specify the engine pool name, engine pool size, but the result is not expected.

They need to remove the subdomain specified before, otherwise, the engine pool will not work.

I need tell customer that, the engine subdomain config has higher priority.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -1652,6 +1652,12 @@ object KyuubiConf {
     .checkValues(EngineType.values.map(_.toString))
     .createWithDefault(EngineType.SPARK_SQL.toString)
 
+  val ENGINE_POOL_FIRST: ConfigEntry[Boolean] = buildConf("kyuubi.engine.pool.first")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1655]: we shall change it to `kyuubi.engine.pool.ignoreSubdomain` and mark it as internal, so the primary users will not care about what's the relationship between engine pool and subdomain

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -1652,6 +1652,14 @@ object KyuubiConf {
     .checkValues(EngineType.values.map(_.toString))
     .createWithDefault(EngineType.SPARK_SQL.toString)
 
+  val ENGINE_POOL_IGNORE_SUBDOMAIN: ConfigEntry[Boolean] =
+    buildConf("kyuubi.engine.pool.ignoreSubdomain")
+      .doc("Whether to ignore subdomain when engine pool conditions are met.")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:1657]: please replace "subdomain" w/ the full conf key

**Review by pan3793 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master

**Pull request closed without merging**